### DISCUSSION TITLE
Changing target_dir based on env ?

### DISCUSSION THREAD

**Discussion by codevogel** [bashly-framework/bashly#525] [Q&A] [answered]
Is it possible to generate the script to a different `target_dir` based on the `env` argument?

I have noticed that `bashly generate --help` lists

```
  -e --env ENV
    Force the generation environment:

    - development   default script output
    - production    remove file marker comments and development functions
```

It would be helpful if I can generate the script in `--env development` to `dev/` and the generate the script in `--env production` to `prod/`.

Or can I perhaps pass a different `bashly-settings.yml` to bashly?

I know I could wrap my `bashly generate` call in a separate script and set the `BASHLY_SETTINGS_PATH` there, but it would be nice if I could pass it as an argument. Am I missing something in the docs for this?

**Comment by DannyBen** [comment, marked as answer]
Well, no there is not really an option like this. Keep in mind that the `target_dir` in the settings is not only for the final script, it is used by other things as well.

Things to consider:

1. There should normally not be a need to generate both versions. Choose one that is best for you and stay with it.
2. You can use ERB in the settings.yml, so you can do something like this
   ```yaml
   target_dir: <%= ENV['BASHLY_ENV'] == 'production' ? '.' : 'dev' %>
   ```
3. Keep in mind that running `bashly generate --env production` does not change the BASHLY_ENV environment variable, so you might need to run something like `BASHLY_ENV=production bashly g`
4. Any setting in the settings.yml file can be [overridden via environment variable](https://bashly.dannyb.co/usage/settings/#environment-variables), so another way could be to do `BASHLY_TARGET_PATH=dev bashly g`
5. If none of the above is acceptable, you can always create a script or a project task runner (see [opcode](https://github.com/DannyBen/opcode), or if you are a Ruby fan - [runfile](https://github.com/dannyben/runfile)) to have a build command that does `bashly generate && mv myscript dev/`

Is this helpful?

**Reply by codevogel** [reply]
ah I completely glossed over the environment variable part. thanks!
